packages/backend/src/services/CsvService/CsvService.ts

```diff
@@ -319,6 +320,84 @@ export class CsvService extends BaseService {
         });
     }
 
+    /**
+     * Stream S3 JSONL data to CSV file
+     * This method is specifically designed to handle JSONL data from S3 storage
+     * and convert it to CSV format.
+     * Unlike streamRowsToFile, this method expects the readStream to be in string format,
+     * not in object mode.
+     */
+    static async streamS3DataRowsToFile(
```

Comment:
[non blocker] I would have named this `streamJsonlRowsToFile`.

Main reason being the method doesn't need to know about 'S3' only 'JSONL'